test/e2e/Makefile

```diff
@@ -0,0 +1,138 @@
+.DEFAULT_GOAL:=help
+
+IMG ?= localhost:5000/cluster-api-provider-cloudstack:latest
+
+GOPROXY := $(shell go env GOPROXY)
+ifeq ($(GOPROXY),)
+GOPROXY := https://proxy.golang.org
+endif
+export GOPROXY
+
+REPO_ROOT := $(shell git rev-parse --show-toplevel)
+
+help:
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z_0-9-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
+
+## --------------------------------------
+## Binaries
+## --------------------------------------
+
+BIN_DIR := $(REPO_ROOT)/bin
+GINKGO := $(BIN_DIR)/ginkgo
+KUSTOMIZE := $(BIN_DIR)/kustomize
+CONTROLLER_GEN := $(BIN_DIR)/controller-gen
+
+$(GINKGO): # Build ginkgo from bin folder.
```

Comment:
Can we consolidate the binary building by just having the main makefile build these?

The main makefile can actually call this one, and in that target, we can have binaries as a prereq for e2e tests.

It would be nice too as we wouldn't need to specify the directory when calling make, so `make <target>` instead of `make --directory=./test/e2e <target>`

Reply:
I tried but I got this error when running e2e test from the main Makefile. 
```
Failed to compile e2e:

main module (github.com/aws/cluster-api-provider-cloudstack) does not contain package github.com/aws/cluster-api-provider-cloudstack/test/e2e
```
I think that Makefiles should be separated as well because test/e2e is a separate module.

Reply:
This error is really just that pwd needs to be the e2e directory.

This can either be done via `cd && <test commands>`, or the main Makefile can call the e2e makefile.

Personally I prefer the one Makefile since I now understand it enough for it to know it can be a single small target.

Reply:
> I think that Makefiles should be separated as well because test/e2e is a separate module.

By this reasoning, we should push this to be a separate repo even.

The one Makefile already manages several modules in this repo.

Reply:
Also, in retrospect, the cd shouldn't be needed. Gingko will run in a specified directory. It just needs to be passed the directory.